src/cli/clioptions.cpp

```diff
@@ -136,44 +137,70 @@ bool parseCLIOptions(QCommandLineParser &parser, QString &errorMessage,
 
   // Validate register initializations
   if (parser.isSet("reginit")) {
-    QStringList regInitList = parser.value("reginit").split(",");
-    for (auto &regInit : regInitList) {
-      QStringList regInitParts = regInit.split("=");
-      if (regInitParts.size() != 2) {
-        errorMessage = "Invalid register initialization '" + regInit +
-                       "' specified (--reginit).";
+    for (const auto &regFileInit : parser.values("reginit")) {
+      if (!regFileInit.contains(':')) {
+        errorMessage = "Cannot find register file type (--reginit).";
         return false;
       }
-      bool ok;
-      int regIdx = regInitParts[0].toInt(&ok);
-      if (!ok) {
-        errorMessage = "Invalid register index '" + regInitParts[0] +
-                       "' specified (--reginit).";
-        return false;
-      }
-
-      auto &vstr = regInitParts[1];
-      VInt regVal;
-      if (vstr.startsWith("0x"))
-        regVal = decodeRadixValue(vstr, Radix::Hex, &ok);
-      else if (vstr.startsWith("0b"))
-        regVal = decodeRadixValue(vstr, Radix::Binary, &ok);
-      else
-        regVal = decodeRadixValue(vstr, Radix::Signed, &ok);
-
-      if (!ok) {
-        errorMessage =
-            "Invalid register value '" + vstr + "' specified (--reginit).";
-        return false;
+      auto regFileSplit = regFileInit.indexOf(':');
+      QString regFile = regFileInit.mid(0, regFileSplit);
+
+      QStringList regInitList = regFileInit.mid(regFileSplit + 1).split(",");
+      for (auto &regInit : regInitList) {
+        QStringList regInitParts = regInit.split("=");
+        if (regInitParts.size() != 2) {
+          errorMessage = "Invalid register initialization '" + regInit +
+                         "' specified (--reginit).";
+          return false;
+        }
+        bool ok;
+        int regIdx = regInitParts[0].toInt(&ok);
+        if (!ok) {
+          errorMessage = "Invalid register index '" + regInitParts[0] +
+                         "' specified (--reginit).";
+          return false;
+        }
+
+        auto &vstr = regInitParts[1];
+        VInt regVal;
+        if (vstr.startsWith("0x"))
+          regVal = decodeRadixValue(vstr, Radix::Hex, &ok);
+        else if (vstr.startsWith("0b"))
+          regVal = decodeRadixValue(vstr, Radix::Binary, &ok);
+        else
+          regVal = decodeRadixValue(vstr, Radix::Signed, &ok);
+
+        if (!ok) {
+          errorMessage =
+              "Invalid register value '" + vstr + "' specified (--reginit).";
+          return false;
+        }
+
+        RegisterFileType regFileType;
+        if (regFile.startsWith("gpr", Qt::CaseInsensitive)) {
```

Comment:
@mortbopet I had to change the `--reginit` command line parameter to accept a register file to index into. Do you think that the names `gpr`/`fpr`/`csr` are fine for specifying which register file to initialize? I feel like the set of register file types should be different for each ISA. For example, I don't think all ISAs have a CSR file.

What do you think about having a different `RegisterFileType` enum for each ISA?

Reply:
How about extending `RegFileInfoInterface` with `RegFileInfoInterface::regFileName()`. Then, the CLI would iterate over the register file names exposed by the ISA, show it if there's a match, and if not, list the possible register file identifiers.

This would then also remove the need for the `RegisterFileType` since the whole point of that is to provide unique identifiers for each register file... might as well move that over to make the registerfile names load bearing 🤔.

Reply:
I think that would work well, thanks!